src/Riok.Mapperly/Descriptors/MappingBuilders/DictionaryMappingBuilder.cs

```diff
@@ -38,6 +42,13 @@ public static class DictionaryMappingBuilder
                 dictionaryObjectFactory);
         }
 
+        // if target is an immutable dictionary then use LinqDictionaryMapper
+        if (SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.ImmutableDictionaryT))
+            return new LinqDicitonaryMapping(ctx.Source, ctx.Target, ResolveStaticMethod(ctx.Types.ImmutableDictionary, ToImmutableDictionaryMethodName)!, keyMapping, valueMapping);
```

Comment:
Instead of asserting non null, I'd add this to the if condition.

Reply:
I assume you want something like:
```C#
if (SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.ImmutableSortedDictionaryT) 
&& ctx.Types.ImmutableSortedDictionary.GetStaticGenericMethod(ToImmutableSortedDictionaryMethodName) is IMethodSymbol method)
            return new LinqDicitonaryMapping(ctx.Source, ctx.Target, method, keyMapping, valueMapping);
```

Currently if the condition is true then we know 100% that it will `ResolveStaticMethod` will not return null. I'm not a fan of  putting logic in a condition; It is misleading because it makes it seem like it could fail. I could change `ResolveStaticMethod` to throw if returning null or keep the nullable suppression operator.